File: easy/0766_toeplitz_matrix/solution.py
class Solution:
    def isToeplitzMatrix(self, matrix: list[list[int]]) -> bool:
        rows = len(matrix)
        cols = len(matrix[0])

        for row in range(rows):
            col = 0
            val = matrix[row][col]

            while row < rows and col < cols:
                if matrix[row][col] != val:
                    return False

                row += 1
                col += 1

        for col in range(1, cols):
            row = 0
            val = matrix[row][col]

            while row < rows and col < cols:
                if matrix[row][col] != val:
                    return False

                row += 1
                col += 1

        return True

File: easy/0766_toeplitz_matrix/solution.cpp
#include <vector>

class Solution {
public:
  bool isToeplitzMatrix(std::vector<std::vector<int>> &matrix) {
    const size_t rows = matrix.size();
    const size_t cols = matrix[0].size();

    for (size_t row = 0; row < rows; ++row) {
      size_t r = row;
      size_t c = 0;

      while (r < rows && c < cols) {
        if (matrix[r++][c++] != matrix[row][0])
          return false;
      }
    }

    for (size_t col = 1; col < cols; ++col) {
      size_t r = 0;
      size_t c = col;

      while (r < rows && c < cols) {
        if (matrix[r++][c++] != matrix[0][col])
          return false;
      }
    }

    return true;
  }
};
